dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFilePersisted.cpp

```diff
@@ -182,17 +177,15 @@ ColumnFilePersisteds deserializeSavedColumnFiles(
             meta.ParseFromString(data),
             "Failed to parse DeltaLayerMeta from string: {}",
             Redact::keyToHexString(data.data(), data.size()));
-        column_files = deserializeSavedColumnFilesInV4Format(context, segment_range, meta);
-        break;
+        return deserializeSavedColumnFilesInV4Format(context, segment_range, meta);
     }
     default:
         throw Exception(
             ErrorCodes::LOGICAL_ERROR,
             "Unexpected delta value version: {}, latest version: {}",
             version,
-            DeltaFormat::V3);
```

Comment:
maybe remove this default branch, so that the compiler can throw error when we miss adding the new branch for DeltaFormat?

Reply:
No.

1. DeltaFormat is not enum type.
2. It is possible that the version is larger than 4 (when data is broken) during deserializing, and we should throw an exception.